static/src/javascripts-legacy/projects/common/modules/email/run-checks.js

```diff
@@ -149,18 +132,21 @@ define([
 
     function allEmailCanRun() {
         var browser = detect.getUserAgent.browser,
-            version = detect.getUserAgent.version;
-
-        return !config.page.shouldHideAdverts &&
-            !config.page.isSensitive &&
-            !emailInserted &&
-            !config.page.isFront &&
-            config.switches.emailInArticle &&
-            !clash.userIsInAClashingAbTest(clash.nonEmailClashingTests) &&
-            storage.session.isAvailable() &&
-            !userHasSeenThisSession() &&
-            !compliantObWidgetIsShown() &&
-            !(browser === 'MSIE' && contains(['7','8','9'], version + ''));
+            version = detect.getUserAgent.version,
+            emailCanRun = !config.page.shouldHideAdverts &&
+                        !config.page.isSensitive &&
+                        !config.page.isFront &&
+                        config.switches.emailInArticle &&
+                        !clash.userIsInAClashingAbTest(clash.nonEmailClashingTests) &&
+                        storage.session.isAvailable() &&
+                        !userHasSeenThisSession() &&
+                        !(browser === 'MSIE' && contains(['7','8','9'], version + ''));
+
+        if (!emailCanRun) {
+            checkMediator.resolveCheck('isEmailInserted', false);
```

Comment:
Without knowing too much about what these changes are doing, I'm a little bit 😢  about adding side effects like this to what is essentially a query function. Does this not belong in the caller, rather than in here?